src/projection/node/create-projection-node.ts

```diff
@@ -1161,6 +1162,27 @@ export function createProjectionNode<I>({
             this.pendingAnimation = sync.update(() => {
                 globalProjectionState.hasAnimatedSinceResize = true
 
+                /**
+                 * Check if this component or any of its ancestors are animating
+                 */
+                if (
+                    process.env.NODE_ENV !== "production" &&
+                    this.options.visualElement
+                ) {
+                    /**
+                     * We need to fire this in a setTimeout because layout animations will start
+                     * before any layout-affecting animations.
+                     */
+                    setTimeout(() => {
+                        warnOnce(
+                            !isTreeAnimatingLayoutAffectingStyle(
+                                this.options.visualElement
+                            ),
+                            `Attempting to animate layout within a component performing an animation on a layout-affecting properties (e.g. "width", "height", "top"). This is likely to break layout animations. Attempt to replace with the layout and style prop, e.g. <motion.div layout style={{ width: 100px }} />.`
```

Comment:
I think logging out the actual div that is at fault here would be helpful for debugging.